app/src/main/java/com/example/dashcarr/presentation/tabs/settings/SettingsFragment.kt

```diff
@@ -13,15 +17,35 @@ class SettingsFragment : BaseFragment<FragmentSettingsBinding>(
     private val viewModel: SettingsViewModel by viewModels()
 
     override fun observeViewModel() {
-        TODO("Not yet implemented")
+        viewModel.logOutState.collectWithLifecycle(viewLifecycleOwner) {
+            findNavController().navigate(R.id.action_global_loginFragment)
+        }
     }
 
     override fun initListeners() {
-        TODO("Not yet implemented")
+        binding.btnLogout.setOnClickListener {
+            viewModel.logOut()
+        }
+        binding.btnProfileSettings.setOnClickListener {
+            //findNavController().navigate(R.id.)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
```

Comment:
remove

Reply:
anyway they gonna be used for those who work with fragment. it's a part of fragments lyfecycle. leave it for now